tests/dragonfly/conftest.py

```diff
@@ -77,7 +77,7 @@ def parse_args(args: List[str]) -> Dict[str, Union[str, None]]:
     return args_dict
 
 
-@pytest.fixture(scope="session", params=[{}])
+@pytest.fixture(scope="function", params=[{}])
```

Comment:
you should change this to function as well

@pytest.fixture(scope="class")
def connection(df_server: DflyInstance):